File: 프로그래머스/lv2/42586. 기능개발/기능개발.py
from collections import deque

def solution(progresses, speeds):
    
    progresses = deque(progresses)
    speeds = deque(speeds)
    
    answer = []
    days = 0
    cnt = 0
    while progresses:
        if progresses[0] + days*speeds[0] >= 100:
            progresses.popleft()
            speeds.popleft()
            cnt += 1
        else:
            if cnt > 0:
                answer.append(cnt)
                cnt = 0
            days += 1
    answer.append(cnt)
        
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    int count = 1;
    
    int next = (100 - progresses[0] + speeds[0] - 1) / speeds[0];
    
    for (int i = 1; i < progresses.size(); i++) {
        int curr = (100 - progresses[i] + speeds[i] - 1) / speeds[i];
        
        if (curr <= next) {
            ++count;
        } else {
            answer.push_back(count);
            count = 1;
            next = curr;
        }
    }
    
    answer.push_back(count);
    
    return answer;
}
